C7Engine/MapUnitExtensions.cs

```diff
@@ -30,19 +31,95 @@ public static void wake(this MapUnit unit)
 		unit.isFortified = false;
 	}
 
+	public static IEnumerable<StrengthBonus> ListStrengthBonusesVersus(this MapUnit unit, MapUnit opponent, bool attacking, bool bombard, TileDirection? attackDirection)
```

Comment:
I'm not really familiar with yield in C#. What's the purpose of it here over just building a list? A comment block explaining would be nice.

Reply:
I'm no C# expert myself. Before C7, the only significant thing I've done with C# was mod Rimworld, and that's where I learned that pattern of a method yield-returning an `IEnumerable`. I like this pattern because it omits the list itself and focuses on its contents, which is what we actually care about. In general it's best to minimize mutable state. Doing things this way is also possibly more efficient since no list needs to be allocated. Though whether that's actually a net win for performance depends on how efficiently the compiler handles yield-returns. I read somewhere that this pattern gets compiled to a state machine, in other words it's not just building a list under the hood.

Reply:
That makes sense. I've been in python land too long to even think about mutability.